HandEyeCalibration/Transform/Transform.cpp

```diff
@@ -0,0 +1,244 @@
+    std::cout << "Axis:\n" << axisAngle.axis() << std::endl;
+    std::cout << "Angle:\n" << axisAngle.angle() << std::endl;
+
+    // Axis-angle to Rotation Vector
+    std::cout << "\nConverting Axis-Angle to Rotation Vector:" << std::endl;
+    if(!axisAngle.axis().isUnitary())
+    {
+        std::cout << "Note! Axis in Axis-Angle pair is not unitary" << std::endl;
+    }
+    robotRepresentations.rotationVector = axisAngle.angle() * axisAngle.axis();
+    std::cout << robotRepresentations.rotationVector << std::endl;
+
+    // Rotation Matrix to Quaternion
+    std::cout << "\nConverting Rotation Matrix to Quaternion:" << std::endl;
+    const Eigen::Quaterniond quaternion(transformationMatrix.linear());
+    robotRepresentations.quaternion = quaternion;
+    std::cout << robotRepresentations.quaternion.coeffs() << std::endl;
+
+    // Rotation Matrix to Roll-Pitch-Yaw
+    for(int i = 0; i < nofRotationConventions; i++)
+    {
+        const RotationConvention convention{ static_cast<RotationConvention>(i) };
+        std::cout << "\nConverting Rotation Matrix to Roll-Pitch-Yaw angles (" << toString(convention)
+                  << "):" << std::endl;
+        robotRepresentations.rotations[i] = { static_cast<RotationConvention>(i),
+                                              rotmat2rpy(transformationMatrix.linear(), convention) };
+        std::cout << robotRepresentations.rotations[i].rollPitchYaw << std::endl;
+    }
+
+    return robotRepresentations;
+}
+
+Eigen::Affine3d transformFromRobotToZivid(const Eigen::Affine3d &transformationMatrix,
+                                          const Representations &representations)
+{
+    // Roll, Pitch and Yaw to Rotation Matrix
+    for(const auto &rotation : representations.rotations)
+    {
+        std::cout << "\nConverting Roll-Pitch-Yaw angles (" << toString(rotation.convention)
+                  << ") to Rotation Matrix:" << std::endl;
+        const Eigen::Matrix3d rotationMatrixFromRollPitchYaw = rpy2rotmat(rotation.rollPitchYaw, rotation.convention);
+        std::cout << rotationMatrixFromRollPitchYaw << std::endl;
+    }
+
+    // Rotation Vector to Axis-angle
+    std::cout << "\nConverting Rotation Vector to Axis-Angle" << std::endl;
+    const Eigen::AngleAxisd axisAngle(representations.rotationVector.norm(),
+                                      representations.rotationVector.normalized());
+    std::cout << "Axis:\n" << axisAngle.axis() << std::endl;
+    std::cout << "Angle:\n" << axisAngle.angle() << std::endl;
+
+    // Axis-Angle to Quaternion
+    std::cout << "\nConverting Axis-Angle to Quaternion:" << std::endl;
+    const Eigen::Quaterniond quaternion(axisAngle);
+    std::cout << quaternion.coeffs() << std::endl;
+
+    // Quaternion to Rotation Matrix
+    std::cout << "\nConverting Quaternion to Rotation Matrix:" << std::endl;
+    auto transformationMatrixFromQuaternion = Eigen::Affine3d(transformationMatrix);
```

Comment:
This was very difficult for me to understand because you skip the step with rotationMatrixFromQuaternion (especially because you use it up with rotationMatrixFromRollPitchYaw). Even though it would add one line of code, I think it would help with the flow and understanding of what is happening. So:

const Eigen::Matrix3d rotationMatrixFromQuaternion = quaternion.toRotationMatrix();

// Roatation Matrix and Translation Vector to Transformation Matrix
const Eigen::Affine3d transformationMatrix;
transformationMatrix.linear() = rotationMatrixFromQuaternion;
transformationMatrix.translation() = translationVector (input translation vector instead of input translation matrix)

Reply:
I'm not against most of it, I do however find it elegant to pass around a reference to Eigen::Affine3d, and access its members, rather than passing references to members.. If everyone agrees I'll change that too.

Reply:
How about this:
```
    // Quaternion to Rotation Matrix
    std::cout << "\nConverting Quaternion to Rotation Matrix:" << std::endl;
    auto transformationMatrixFromQuaternion = Eigen::Affine3d();
    transformationMatrixFromQuaternion.linear() = quaternion.toRotationMatrix();
    transformationMatrixFromQuaternion.translation() = transformationMatrix.translation();
    std::cout << transformationMatrixFromQuaternion.linear() << std::endl;
```

Reply:
This is easier to understand. There is still a difference between what the comment states and what is done within the 4 lines of code below it. It would be even more intuitive if you could split a) conversion from quat to rot mat and b) combining rot mat and trans vec to 4x4 transform as two sections. However, let's not get bogged down with nitpicks. What I suggest is most likely not the best way when it comes to good programming. I was more focused on the educational aspect. I am also used to using this http://www.petercorke.com/RTB/r9/html/rt2tr.html

Hear my input and chose what you think is best, I am ok with it.